// react-fabric/lib/ChoiceField/ChoiceField.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Label = require('../Label');

var _Label2 = _interopRequireDefault(_Label);

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _isDefined = require('../util/isDefined.js');

var _isDefined2 = _interopRequireDefault(_isDefined);

var _ChoiceField = require('./ChoiceField.scss');

var _ChoiceField2 = _interopRequireDefault(_ChoiceField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ChoiceField = function ChoiceField(_ref) {
  var className = _ref.className;
  var disabled = _ref.disabled;
  var id = _ref.id;
  var label = _ref.label;
  var name = _ref.name;
  var required = _ref.required;
  var type = _ref.type;
  var checked = _ref.checked;
  var props = (0, _objectWithoutProperties3.default)(_ref, ['className', 'disabled', 'id', 'label', 'name', 'required', 'type', 'checked']);

  var inputId = 'ChoiceField_' + (id || name || Date.now()) + '_input';

  return _react2.default.createElement(
    'div',
    { 'data-fabric': 'ChoiceField',
      id: id,
      className: className,
      styleName: 'ms-ChoiceField' },
    _react2.default.createElement('input', (0, _extends3.default)({}, props, {
      children: null,
      id: inputId,
      name: name,
      type: type,
      disabled: disabled,
      checked: (0, _isDefined2.default)(checked) ? checked : false,
      styleName: 'ms-ChoiceField-input' })),
    _react2.default.createElement(
      'label',
      { styleName: 'ms-ChoiceField-field',
        htmlFor: inputId },
      _react2.default.createElement(
        _Label2.default,
        { componentClass: 'span', styleName: 'ms-Label', required: required },
        label
      )
    )
  );
};
ChoiceField.displayName = 'ChoiceField';
ChoiceField.propTypes = {
  checked: _react2.default.PropTypes.bool,
  className: _react2.default.PropTypes.string,
  disabled: _react2.default.PropTypes.bool,
  id: _react2.default.PropTypes.string,
  label: _react2.default.PropTypes.node,
  name: _react2.default.PropTypes.string.isRequired,
  required: _react2.default.PropTypes.bool,
  type: _react2.default.PropTypes.oneOf(['radio', 'checkbox']).isRequired
};
ChoiceField.defaultProps = {
  disabled: false,
  require: false,
  type: 'checkbox'
};

exports.default = (0, _fabricComponent2.default)(ChoiceField, _ChoiceField2.default);
