// react-fabric/lib/ChoiceFieldGroup/ChoiceFieldGroup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Label = require('../Label');

var _Label2 = _interopRequireDefault(_Label);

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _ChoiceField = require('../ChoiceField/ChoiceField.scss');

var _ChoiceField2 = _interopRequireDefault(_ChoiceField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var handleChildChanged = function handleChildChanged(onChange) {
  return function (e) {
    if (onChange && e.target.checked) {
      onChange(e);
    }
  };
};

var createChoices = function createChoices(_ref) {
  var children = _ref.children;
  var name = _ref.name;
  var value = _ref.value;
  var onChange = _ref.onChange;
  return _react2.default.Children.map(children, function (child, i) {
    if (child === null) {
      return child;
    }

    return _react2.default.cloneElement(child, {
      type: 'radio',
      name: name,
      id: 'ChoiceFieldGroup_' + name + '[' + i + ']',
      checked: value === child.props.value,
      onChange: handleChildChanged(onChange)
    });
  });
};

var ChoiceFieldGroup = function ChoiceFieldGroup(_ref2) {
  var children = _ref2.children;
  var name = _ref2.name;
  var required = _ref2.required;
  var title = _ref2.title;
  var value = _ref2.value;
  var onChange = _ref2.onChange;
  var props = (0, _objectWithoutProperties3.default)(_ref2, ['children', 'name', 'required', 'title', 'value', 'onChange']);
  return _react2.default.createElement(
    'div',
    (0, _extends3.default)({ 'data-fabric': 'ChoiceFieldGroup'
    }, props, {
      styleName: 'ms-ChoiceFieldGroup' }),
    _react2.default.createElement(
      'div',
      null,
      title && _react2.default.createElement(
        _Label2.default,
        { required: required },
        title
      )
    ),
    createChoices({ children: children, name: name, value: value, onChange: onChange })
  );
};
ChoiceFieldGroup.propTypes = {
  children: _react2.default.PropTypes.node,
  className: _react2.default.PropTypes.string,
  name: _react2.default.PropTypes.string.isRequired,
  required: _react2.default.PropTypes.bool,
  title: _react2.default.PropTypes.string,
  value: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func
};
ChoiceFieldGroup.defaultProps = {
  required: false
};

exports.default = (0, _fabricComponent2.default)(ChoiceFieldGroup, _ChoiceField2.default);
