// react-fabric/lib/Label/Label.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _elementType = require('react-prop-types/lib/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

var _invokeWhenNotDisabled = require('../util/invokeWhenNotDisabled.js');

var _invokeWhenNotDisabled2 = _interopRequireDefault(_invokeWhenNotDisabled);

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _Label = require('./Label.scss');

var _Label2 = _interopRequireDefault(_Label);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Label = function Label(_ref) {
  var children = _ref.children;
  var ComponentClass = _ref.componentClass;
  var disabled = _ref.disabled;
  var htmlFor = _ref.htmlFor;
  var onClick = _ref.onClick;
  var required = _ref.required;
  var props = (0, _objectWithoutProperties3.default)(_ref, ['children', 'componentClass', 'disabled', 'htmlFor', 'onClick', 'required']);
  return _react2.default.createElement(
    ComponentClass,
    (0, _extends3.default)({ 'data-fabric': 'Label'
    }, props, {
      htmlFor: htmlFor,
      onClick: (0, _invokeWhenNotDisabled2.default)(disabled, onClick),
      styleName: (0, _classnames2.default)('ms-Label', {
        'is-required': required,
        'is-disabled': disabled
      }) }),
    children
  );
};
Label.displayName = 'Label';
Label.propTypes = {
  children: _react2.default.PropTypes.node,
  componentClass: _elementType2.default,
  disabled: _react2.default.PropTypes.bool,
  htmlFor: _react2.default.PropTypes.string,
  onClick: _react2.default.PropTypes.func,
  required: _react2.default.PropTypes.bool
};
Label.defaultProps = {
  componentClass: 'label',
  disabled: false,
  required: false
};

exports.default = (0, _fabricComponent2.default)(Label, _Label2.default);
