// react-fabric/lib/fabricComponent/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isFabricComponent = undefined;

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

exports.configure = configure;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactCssModules = require('react-css-modules');

var _reactCssModules2 = _interopRequireDefault(_reactCssModules);

var _lodash = require('lodash.isfunction');

var _lodash2 = _interopRequireDefault(_lodash);

var _lodash3 = require('lodash.identity');

var _lodash4 = _interopRequireDefault(_lodash3);

var _hoistNonReactStatics = require('hoist-non-react-statics');

var _hoistNonReactStatics2 = _interopRequireDefault(_hoistNonReactStatics);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_CONFIGURATION = {
  preprocessStyles: _lodash4.default
};

var configuration = (0, _extends3.default)({}, DEFAULT_CONFIGURATION);

// TODO: move to readme / docs
// const prependGlobalClassNames = styles => (
//   Object.keys(styles).reduce((newStyles, style) => {
//     newStyles[style] = `${style} ${styles[style]}` // eslint-disable-line no-param-reassign
//
//     return newStyles
//   }, {})
// )

var higherOrderFunction = function higherOrderFunction(Component) {
  var styles = arguments.length <= 1 || arguments[1] === undefined ? {} : arguments[1];
  var options = arguments.length <= 2 || arguments[2] === undefined ? {} : arguments[2];

  var fabricStyles = configuration.preprocessStyles(styles);
  var fabricOptions = (0, _extends3.default)({}, options, {
    allowMultiple: true
  });

  var WrappedComponent = (0, _reactCssModules2.default)(Component, fabricStyles, fabricOptions);
  var FabricComponent = function FabricComponent(_ref) {
    var ignoreStyles = _ref.styles;
    var props = (0, _objectWithoutProperties3.default)(_ref, ['styles']);
    return _react2.default.createElement(WrappedComponent, props);
  };

  var componentDisplayName = Component.displayName || Component.name;

  FabricComponent.displayName = 'FabricComponent(' + componentDisplayName + ')';
  FabricComponent.WrappedComponent = Component;

  (0, _hoistNonReactStatics2.default)(FabricComponent, Component);

  return FabricComponent;
};

var decorator = function decorator() {
  for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }

  return function (Component) {
    return higherOrderFunction.apply(undefined, [Component].concat(args));
  };
};

exports.default = function () {
  return (0, _lodash2.default)(arguments.length <= 0 ? undefined : arguments[0]) ? higherOrderFunction.apply(undefined, arguments) : decorator.apply(undefined, arguments);
};

var isFabricComponent = exports.isFabricComponent = function isFabricComponent() {
  for (var _len2 = arguments.length, componentTypes = Array(_len2 > 1 ? _len2 - 1 : 0), _key2 = 1; _key2 < _len2; _key2++) {
    componentTypes[_key2 - 1] = arguments[_key2];
  }

  var component = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

  var type = component.type || {};

  return componentTypes.indexOf(type) !== -1;
};

function configure(options) {
  configuration = (0, _extends3.default)({}, DEFAULT_CONFIGURATION, {
    options: options
  });
}
