// react-fabric/lib/Button/Button.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _onlyWhen = require('../propTypes/onlyWhen');

var _onlyWhen2 = _interopRequireDefault(_onlyWhen);

var _Icon = require('../Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _propTypes = require('../propTypes');

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _invokeWhenNotDisabled = require('../util/invokeWhenNotDisabled.js');

var _invokeWhenNotDisabled2 = _interopRequireDefault(_invokeWhenNotDisabled);

var _Button = require('./Button.scss');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BUTTON_TYPES = ['command', 'compound', 'hero', 'primary'];

var Button = function Button(_ref) {
  var children = _ref.children;
  var description = _ref.description;
  var disabled = _ref.disabled;
  var glyph = _ref.glyph;
  var onClick = _ref.onClick;
  var submit = _ref.submit;
  var type = _ref.type;
  var props = (0, _objectWithoutProperties3.default)(_ref, ['children', 'description', 'disabled', 'glyph', 'onClick', 'submit', 'type']);
  return _react2.default.createElement(
    'button',
    (0, _extends3.default)({ 'data-fabric': 'Button'
    }, props, {
      styleName: (0, _classnames2.default)('ms-Button', (0, _defineProperty3.default)({
        'is-disabled': disabled
      }, 'ms-Button--' + type, type)),
      type: submit ? 'submit' : 'button',
      onClick: (0, _invokeWhenNotDisabled2.default)(disabled, onClick) }),
    _react2.default.createElement(
      'span',
      { styleName: 'ms-Button-icon' },
      glyph && _react2.default.createElement(_Icon2.default, { styleName: 'ms-Icon', glyph: glyph })
    ),
    _react2.default.createElement(
      'span',
      { styleName: 'ms-Button-label' },
      children
    ),
    _react2.default.createElement(
      'span',
      { styleName: 'ms-Button-description' },
      description
    )
  );
};
Button.displayName = 'Button';
Button.propTypes = {
  children: _react2.default.PropTypes.node,
  description: (0, _onlyWhen2.default)(function (_ref2) {
    var type = _ref2.type;
    return type === 'compound';
  }, 'type is compound', _react2.default.PropTypes.node),
  disabled: _react2.default.PropTypes.bool,
  glyph: _propTypes.glyphType,
  onClick: _react2.default.PropTypes.func,
  submit: _react2.default.PropTypes.bool,
  type: _react2.default.PropTypes.oneOf(BUTTON_TYPES)
};

exports.default = (0, _fabricComponent2.default)(Button, _Button2.default);
