// react-fabric/lib/TextField/TextField.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Label = require('../Label');

var _Label2 = _interopRequireDefault(_Label);

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _isDefined = require('../util/isDefined.js');

var _isDefined2 = _interopRequireDefault(_isDefined);

var _TextField = require('./TextField.scss');

var _TextField2 = _interopRequireDefault(_TextField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TextField = function TextField(_ref) {
  var active = _ref.active;
  var className = _ref.className;
  var description = _ref.description;
  var disabled = _ref.disabled;
  var id = _ref.id;
  var label = _ref.label;
  var multiline = _ref.multiline;
  var name = _ref.name;
  var placeholder = _ref.placeholder;
  var required = _ref.required;
  var underlined = _ref.underlined;
  var value = _ref.value;
  var props = (0, _objectWithoutProperties3.default)(_ref, ['active', 'className', 'description', 'disabled', 'id', 'label', 'multiline', 'name', 'placeholder', 'required', 'underlined', 'value']);

  var inputId = 'TextField_' + (id || name || Date.now()) + '_input';
  var input = _react2.default.createElement(multiline ? 'textarea' : 'input', (0, _extends3.default)({}, props, {
    active: active,
    disabled: disabled,
    id: inputId,
    name: name,
    styleName: 'ms-TextField-field',
    value: (0, _isDefined2.default)(value) ? value : ''
  }));
  var showLabel = placeholder ? !active && !value : true;

  return _react2.default.createElement(
    'div',
    { 'data-fabric': 'TextField',
      id: id,
      className: className,
      styleName: (0, _classnames2.default)('ms-TextField', {
        'ms-TextField--underlined': underlined,
        'ms-TextField--multiline': multiline,
        'ms-TextField--placeholder': placeholder,
        'is-disabled': disabled,
        'is-required': required
      }) },
    label && showLabel && _react2.default.createElement(
      _Label2.default,
      { styleName: 'ms-Label', htmlFor: inputId },
      label
    ),
    input,
    description && _react2.default.createElement(
      'span',
      { styleName: 'ms-TextField-description' },
      description
    )
  );
};
TextField.displayName = 'TextField';
TextField.propTypes = {
  active: _react2.default.PropTypes.bool,
  className: _react2.default.PropTypes.string,
  description: _react2.default.PropTypes.node,
  disabled: _react2.default.PropTypes.bool,
  id: _react2.default.PropTypes.string,
  label: _react2.default.PropTypes.node,
  multiline: _react2.default.PropTypes.bool,
  name: _react2.default.PropTypes.string,
  placeholder: _react2.default.PropTypes.bool,
  required: _react2.default.PropTypes.bool,
  underlined: _react2.default.PropTypes.bool,
  value: _react2.default.PropTypes.string
};
TextField.defaultProps = {
  active: false,
  disabled: false,
  multine: false,
  placeholder: false,
  required: false,
  underlined: false
};

exports.default = (0, _fabricComponent2.default)(TextField, _TextField2.default);
