// react-fabric/lib/SearchBox/SearchBox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _class, _temp2, _initialiseProps;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Icon = require('../Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _SearchBox = require('./SearchBox.scss');

var _SearchBox2 = _interopRequireDefault(_SearchBox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SearchBox = (_temp2 = _class = function (_React$Component) {
  (0, _inherits3.default)(SearchBox, _React$Component);

  function SearchBox() {
    var _Object$getPrototypeO;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, SearchBox);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_Object$getPrototypeO = (0, _getPrototypeOf2.default)(SearchBox)).call.apply(_Object$getPrototypeO, [this].concat(args))), _this), _initialiseProps.call(_this), _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(SearchBox, [{
    key: 'render',
    value: function render() {
      var _props = this.props;
      var active = _props.active;
      var className = _props.className;
      var onChange = _props.onChange;
      var placeholder = _props.placeholder;
      var value = _props.value;
      var props = (0, _objectWithoutProperties3.default)(_props, ['active', 'className', 'onChange', 'placeholder', 'value']);


      return _react2.default.createElement(
        'div',
        (0, _extends3.default)({ 'data-fabric': 'SearchBox'
        }, props, {
          onMouseOver: this.handleMouseOver.bind(this),
          onMouseOut: this.handleMouseOut.bind(this),
          className: className,
          styleName: (0, _classnames2.default)('ms-SearchBox', {
            'is-active': active
          }) }),
        _react2.default.createElement('input', (0, _extends3.default)({}, props, {
          value: value,
          onChange: onChange,
          styleName: (0, _classnames2.default)('ms-SearchBox-field', {
            'hovering': this.state.hover
          }) })),
        !value && _react2.default.createElement(
          'label',
          { styleName: 'ms-SearchBox-label' },
          _react2.default.createElement(_Icon2.default, { styleName: 'ms-SearchBox-icon', glyph: 'search' }),
          placeholder
        ),
        _react2.default.createElement(
          'button',
          { styleName: 'ms-SearchBox-closeButton', onMouseDown: onChange },
          _react2.default.createElement(_Icon2.default, { glyph: 'x' })
        )
      );
    }
  }]);
  return SearchBox;
}(_react2.default.Component), _class.displayName = 'SearchBox', _class.propTypes = {
  active: _react2.default.PropTypes.bool,
  className: _react2.default.PropTypes.string,
  onChange: _react2.default.PropTypes.func,
  onMouseOver: _react2.default.PropTypes.func,
  onMouseOut: _react2.default.PropTypes.func,
  placeholder: _react2.default.PropTypes.string,
  value: _react2.default.PropTypes.string
}, _class.defaultProps = {
  active: false,
  placeholder: 'Search',
  value: ''
}, _initialiseProps = function _initialiseProps() {
  var _this2 = this;

  this.state = { hover: false };

  this.handleMouseOver = function () {
    _this2.setState((0, _extends3.default)({}, _this2.state, {
      hover: true
    }));

    if (_this2.props.onMouseOver) {
      var _props2;

      (_props2 = _this2.props).onMouseOver.apply(_props2, arguments);
    }
  };

  this.handleMouseOut = function () {
    _this2.setState((0, _extends3.default)({}, _this2.state, {
      hover: false
    }));

    if (_this2.props.onMouseOut) {
      var _props3;

      (_props3 = _this2.props).onMouseOut.apply(_props3, arguments);
    }
  };
}, _temp2);
exports.default = (0, _fabricComponent2.default)(SearchBox, _SearchBox2.default);
