// react-fabric/lib/Dialog/DialogButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Icon = require('../Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _Dialog = require('./Dialog.scss');

var _Dialog2 = _interopRequireDefault(_Dialog);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DialogButton = function DialogButton(_ref) {
  var children = _ref.children;
  var className = _ref.className;
  var close = _ref.close;
  var onClick = _ref.onClick;
  return _react2.default.createElement(
    'button',
    { onClick: onClick,
      styleName: (0, _classnames2.default)('ms-Dialog-button', {
        'ms-Dialog-button--close': close
      }),
      className: className },
    close && _react2.default.createElement(_Icon2.default, { styleName: 'ms-Icon ms-Icon--x', glyph: 'x' }),
    children
  );
};
DialogButton.propTypes = {
  children: _react2.default.PropTypes.node,
  className: _react2.default.PropTypes.string,
  close: _react2.default.PropTypes.bool,
  onClick: _react2.default.PropTypes.func
};

exports.default = (0, _fabricComponent2.default)(DialogButton, _Dialog2.default);
