// react-fabric/lib/Dialog/Dialog.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _DialogButton = require('./DialogButton.js');

var _DialogButton2 = _interopRequireDefault(_DialogButton);

var _DialogTitle = require('./DialogTitle.js');

var _DialogTitle2 = _interopRequireDefault(_DialogTitle);

var _DialogSubtext = require('./DialogSubtext.js');

var _DialogSubtext2 = _interopRequireDefault(_DialogSubtext);

var _DialogAction = require('./DialogAction.js');

var _DialogAction2 = _interopRequireDefault(_DialogAction);

var _Overlay = require('../Overlay');

var _Overlay2 = _interopRequireDefault(_Overlay);

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _Dialog = require('./Dialog.scss');

var _Dialog2 = _interopRequireDefault(_Dialog);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_TITLE = _react2.default.createElement(
  _DialogTitle2.default,
  null,
  'Dialog'
);

var scanChildren = function scanChildren(children) {
  return _react2.default.Children.toArray(children).reduce(function (r, child) {
    if ((0, _fabricComponent.isFabricComponent)(child, _DialogTitle2.default)) {
      r.title = r.title === DEFAULT_TITLE ? child : r.title; // eslint-disable-line no-param-reassign
    } else if ((0, _fabricComponent.isFabricComponent)(child, _DialogAction2.default)) {
        r.actions.push(child);
      } else {
        r.content.push(child);
      }
    return r;
  }, { title: DEFAULT_TITLE, content: [], actions: [] });
};

var Dialog = function Dialog(_ref) {
  var children = _ref.children;
  var closeable = _ref.closeable;
  var darkOverlay = _ref.darkOverlay;
  var hidden = _ref.hidden;
  var onClose = _ref.onClose;
  var props = (0, _objectWithoutProperties3.default)(_ref, ['children', 'closeable', 'darkOverlay', 'hidden', 'onClose']);

  var _scanChildren = scanChildren(children);

  var title = _scanChildren.title;
  var content = _scanChildren.content;
  var actions = _scanChildren.actions;


  return _react2.default.createElement(
    'div',
    (0, _extends3.default)({ 'data-fabric': 'Dialog'
    }, props, {
      styleName: (0, _classnames2.default)('ms-Dialog', {
        'ms-Dialog--close': closeable,
        'ms-Dialog--hidden': hidden
      }) }),
    _react2.default.createElement(_Overlay2.default, { styleName: 'ms-Overlay', dark: darkOverlay, onClick: closeable ? onClose : null }),
    _react2.default.createElement(
      'div',
      { styleName: 'ms-Dialog-main' },
      closeable && _react2.default.createElement(_DialogButton2.default, { close: true, onClick: onClose }),
      title,
      _react2.default.createElement(
        'div',
        { styleName: 'ms-Dialog-inner' },
        _react2.default.createElement(
          'div',
          { styleName: 'ms-Dialog-content' },
          content
        ),
        actions.length > 0 && _react2.default.createElement(
          'div',
          { styleName: 'ms-Dialog-actions' },
          _react2.default.createElement(
            'div',
            { styleName: 'ms-Dialog-actionsRight' },
            actions
          )
        )
      )
    )
  );
};
Dialog.displayName = 'Dialog';
Dialog.propTypes = {
  children: _react2.default.PropTypes.node,
  closeable: _react2.default.PropTypes.bool,
  darkOverlay: _react2.default.PropTypes.bool,
  hidden: _react2.default.PropTypes.bool,
  onClose: _react2.default.PropTypes.func
};
Dialog.defaultProps = {
  closeable: false,
  darkOverlay: false,
  hidden: true
};
Dialog.Subtext = _DialogSubtext2.default;
Dialog.Action = _DialogAction2.default;
Dialog.Title = _DialogTitle2.default;

exports.default = (0, _fabricComponent2.default)(Dialog, _Dialog2.default);
