// react-fabric/lib/NavBar/NavBar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _dec, _class, _class2, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Icon = require('../Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _NavBarItem = require('./NavBarItem.js');

var _NavBarItem2 = _interopRequireDefault(_NavBarItem);

var _NavBarLink = require('./NavBarLink.js');

var _NavBarLink2 = _interopRequireDefault(_NavBarLink);

var _NavBarTitle = require('./NavBarTitle.js');

var _NavBarTitle2 = _interopRequireDefault(_NavBarTitle);

var _fabricComponent = require('../fabricComponent');

var _fabricComponent2 = _interopRequireDefault(_fabricComponent);

var _NavBar = require('./NavBar.scss');

var _NavBar2 = _interopRequireDefault(_NavBar);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var scanChildren = function scanChildren(children) {
  return _react2.default.Children.toArray(children).reduce(function (r, child) {
    if ((0, _fabricComponent.isFabricComponent)(child, _NavBarTitle2.default)) {
      r.title = child; // eslint-disable-line no-param-reassign
    } else if ((0, _fabricComponent.isFabricComponent)(child, _NavBarItem2.default, _NavBarLink2.default)) {
        r.items.push(child);
      } else {
        r.items.push(_react2.default.createElement(
          _NavBarItem2.default,
          { key: child.key },
          child
        ));
      }

    return r;
  }, { title: null, items: [] });
};

var NavBar = (_dec = (0, _fabricComponent2.default)(_NavBar2.default), _dec(_class = (_temp = _class2 = function (_React$Component) {
  (0, _inherits3.default)(NavBar, _React$Component);

  function NavBar() {
    (0, _classCallCheck3.default)(this, NavBar);
    return (0, _possibleConstructorReturn3.default)(this, (0, _getPrototypeOf2.default)(NavBar).apply(this, arguments));
  }

  (0, _createClass3.default)(NavBar, [{
    key: '_closeMenu',
    value: function _closeMenu() {
      var _props = this.props;
      var closeMenu = _props.closeMenu;
      var isMenuOpen = _props.isMenuOpen;


      if (isMenuOpen) {
        if (closeMenu) {
          closeMenu();
        }
      }
    }
  }, {
    key: '_openMenu',
    value: function _openMenu() {
      var openMenu = this.props.openMenu;


      if (openMenu) {
        openMenu();
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props2 = this.props;
      var children = _props2.children;
      var isMenuOpen = _props2.isMenuOpen;
      var props = (0, _objectWithoutProperties3.default)(_props2, ['children', 'isMenuOpen']);

      var _scanChildren = scanChildren(children);

      var title = _scanChildren.title;
      var items = _scanChildren.items;

      var styleName = (0, _classnames2.default)('ms-NavBar', {
        'is-open': isMenuOpen
      });

      return _react2.default.createElement(
        'div',
        (0, _extends3.default)({ 'data-fabric': 'NavBar'
        }, props, {
          onClick: this._closeMenu.bind(this),
          styleName: styleName }),
        items.length > 0 && _react2.default.createElement(
          'div',
          { onClick: this._openMenu.bind(this),
            styleName: (0, _classnames2.default)('ms-NavBar-openMenu', {
              'is-open': isMenuOpen
            }) },
          _react2.default.createElement(_Icon2.default, { glyph: 'menu' })
        ),
        title,
        _react2.default.createElement(
          'ul',
          { styleName: 'ms-NavBar-items' },
          items
        )
      );
    }
  }]);
  return NavBar;
}(_react2.default.Component), _class2.propTypes = {
  title: _react2.default.PropTypes.node,
  children: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.element, _react2.default.PropTypes.arrayOf(function (propValue, key, componentName /* , location, propFullName */) {
    // if (!isFabricComponent(propValue[key], NavBarItem, NavBarLink, NavBarTitle)) {
    //   return new Error(
    //     `Invalid prop '${propFullName}' supplied to '${componentName}'. Must be one of
    //     NavBar.Item, NavBar.Link or NavBar.Title`
    //   )
    // }
    if (propValue.filter(function (value) {
      return (0, _fabricComponent.isFabricComponent)(value, _NavBarTitle2.default);
    }).length > 1) {
      return new Error('Multiple NavBar.Title props supplied to \'' + componentName + '\'. Maximal one is allowed.');
    }
    return null;
  })]),
  openMenu: _react2.default.PropTypes.func,
  closeMenu: _react2.default.PropTypes.func,
  isMenuOpen: _react2.default.PropTypes.bool,
  styles: _react2.default.PropTypes.object
}, _class2.defaultProps = {
  isMenuOpen: false
}, _temp)) || _class);

NavBar.Item = _NavBarItem2.default;
NavBar.Link = _NavBarLink2.default;
NavBar.Title = _NavBarTitle2.default;

exports.default = NavBar;
